mender.go

```diff
@@ -409,14 +411,34 @@ func (m *mender) Authorize() menderError {
 
 	rsp, err := m.authReq.Request(m.api, m.config.ServerURL, m.authMgr)
 	if err != nil {
-		errCause := errors.Cause(err)
-		if errCause == client.AuthErrorUnauthorized {
-			// make sure to remove auth token once device is rejected
-			if remErr := m.authMgr.RemoveAuthToken(); remErr != nil {
-				log.Warn("can not remove rejected authentication token")
+		// If (multiple) failover servers are available, cycle through
+		// list and try to authorize.
+		nextServer := nextServerIterator(m)
+		if nextServer != nil {
+			var srv string
+			log.Warnf("Authorization resulted in error: %s", err)
+			log.Warn("Trying to authorize on failover server(s).")
+			for srv = nextServer(); srv != ""; srv = nextServer() {
+				rsp, err = m.authReq.Request(m.api, m.config.ServerURL, m.authMgr)
```

Comment:
Here is an example of where I think the logic could be simplified if only using one single list: You are calling `Request` here, but it's already called once above as well. My thinking is that this could be reduced to simply entering the loop immediately, before you have called any `Request` function, and then start with the first entry, which is the primary server. Does that make sense?

Reply:
So you're saying that we should iterate the list for each request instead of completely shifting attention from one server to another on an unsuccessful request?

Reply:
Well, aren't those two things kind of the same? Try one server, if it fails, try next server. So yes, I guess that's what I'm saying.